winforms/src/toga_winforms/widgets/textinput.py

```diff
@@ -74,37 +73,20 @@ def set_color(self, color):
         else:
             self.native.ForeColor = self.native.DefaultForeColor
 
-    def set_background_color(self, value):
-        if value:
-            self.native.BackColor = native_color(value)
-        else:
-            self.native.BackColor = native_color(TRANSPARENT)
-
     def rehint(self):
         # Height of a text input is known and fixed.
         # Width must be > 100
         # print("REHINT TextInput", self, self.native.PreferredSize)
         self.interface.intrinsic.width = at_least(self.interface._MIN_WIDTH)
         self.interface.intrinsic.height = self.native.PreferredSize.Height
 
-    def set_on_change(self, handler):
-        # No special handling required
-        pass
-
-    def set_on_gain_focus(self, handler):
-        # No special handling required
-        pass
-
-    def set_on_lose_focus(self, handler):
-        # No special handling required
-        pass
-
     def winforms_text_changed(self, sender, event):
         self.interface.on_change(self.interface)
         self.interface.validate()
 
-    def winforms_validated(self, sender, event):
-        self.interface.validate()
```

Comment:
I can't work out what the Validated event handler was supposed to do, but everything seems to work fine without it.

Reply:
Winforms provides explicit event hooks to perform validation independent of focus and text change; I think the intention is that you can prevent a widget from losing focus if it's not currently valid. However, we're not blocking focus in that way, so I guess we can just piggyback on the change event.